hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/stats/SidecarInstrumentationImpl.java

```diff
@@ -0,0 +1,208 @@
+        createAccumulators();
+    }
+
+    /** Create an accumulator for each kind of sidecar record */
+    private void createAccumulators() {
+        for (final var e : SidecarType.values())
+            accumulators.put(e, new SidecarAccumulators(e.name()));
+    }
+
+    /**
+     * Add a sample of the serialized size of a sidecar transaction, for a particular kind of
+     * sidecar
+     */
+    @Override
+    public void addSample(@NonNull final SidecarType type, long size) {
+        final var acc = accumulators.get(type);
+        acc.serializedSize().addSample(size);
+    }
+
+    /**
+     * Add a sample of the compute duration of a sidecar transaction, for a particular kind of
+     * sidecar
+     */
+    @Override
+    public void addSample(@NonNull final SidecarType type, @NonNull final Duration duration) {
+        final var acc = accumulators.get(type);
+        acc.computeDuration().addSample(duration.toNanos());
+    }
+
+    /**
+     * Add a sample of both serialized size and compute time for a sidecar transaction, for a
+     * particular kind of sidecar
+     */
+    @Override
+    public void addSample(
+            @NonNull final SidecarType type, long size, @NonNull final Duration duration) {
+        final var acc = accumulators.get(type);
+        acc.serializedSize().addSample(size);
+        acc.computeDuration().addSample(duration.toNanos());
+    }
+
+    /**
+     * Add the entire accumulated values of a SidecarInstrumentation to this one (which is a summary
+     * of multiple records
+     */
+    @SuppressWarnings("java:S2445")
+    // sync on method parameter _OK_ because adding samples goes only in _one_ direction ever (from
+    // more specific accumulator to more general accumulator)
+    @Override
+    public void addSamples(@NonNull final SidecarInstrumentation samples) {
+        if (samples instanceof SidecarInstrumentationImpl impl) {
+            synchronized (this) {
+                synchronized (samples) {
+                    for (final var e : SidecarType.values()) {
+                        final var thisAcc = accumulators.get(e);
+                        final var thatAcc = impl.accumulators.get(e);
+                        thisAcc.computeDuration().addSamples(thatAcc.computeDuration());
+                        thisAcc.serializedSize().addSamples(thatAcc.serializedSize());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Start timing a split. Use the returned record in a try-with-resources that encloses the
+     * computation to be timed.
+     */
+    @Override
+    public Closeable startTimer(@NonNull final SidecarType type) {
+        return new DurationTimer(this, type, clock.instant());
+    }
+
+    /**
+     * Measures a duration split, adding the elapsed duration to the given accumulator when
+     * `close()` is called (probably by a try-with-resources statement).
+     */
+    private record DurationTimer(
```

Comment:
Again I would prefer to use `class`.

Reply:
Yet this would be a perfect place for the simplicity of a record (were it not that I could have used a lambda instead as you point out next!)  No need for constructors or field definitions for this simple callback ...